provider/spark.go

```diff
@@ -944,40 +972,91 @@ func (s *S3Materialization) NumRows() (int64, error) {
 }
 
 func (s *S3Materialization) IterateSegment(begin, end int64) (FeatureIterator, error) {
-	stream, err := s.store.RowStreamFromSelectQuery(s.Key, fmt.Sprintf("SELECT * FROM S3Object WHERE row_number > %d AND row_number <= %d", begin, end))
+	stream, err := s.store.ResourceStreamConv(s.Key)
 	if err != nil {
 		return nil, err
 	}
+	for i := int64(0); i < begin; i++ {
```

Comment:
Is this the only way to ignore the earlier values?

Reply:
The parquet reader has a skip rows function. replaced this with that implementation